consensus/src/pipeline/monitor.rs

```diff
@@ -12,23 +15,30 @@ use std::{
     time::{Duration, Instant},
 };
 
+const MONITOR: &str = "consensus-monitor";
+
 pub struct ConsensusMonitor {
+    // TODO: change the termination process using a chanel instead so we can (biased) select in the worker
```

Comment:
You can have `tick_service.tick` return the reason for breaking and avoid the terminate boolean here altogether -- simply exit the monitor when tick reason is shutdown